src/boot/efi/fuzz-bcd.c

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "alloc-util.h"
+#include "fuzz.h"
+#include "utf8.h"
+
+#include "bcd.c"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+        /* This limit was borrowed from src/boot/efi/boot.c
+         * I'm not sure where it came from initially though
+         */
+        if (size > 100*1024)
+                return 0;
+
+        _cleanup_free_ void *p = memdup(data, size);
+        assert_se(p);
+
+        char16_t *title = get_bcd_title(p, size);
```

Comment:
`get_bcd_title` isn't UBSan-friendly because it triggers a lot of "unaligned access" reports. I'm not sure what to do about it. If it's supposed to be run on x86 platforms only probably it shouldn't be even be built on non-x86 architectures: https://github.com/systemd/systemd/pull/21727#issuecomment-1001624144

Reply:
It was fixed in https://github.com/systemd/systemd/pull/21922